src/cpu/aarch64/acl_pooling.hpp

```diff
@@ -0,0 +1,241 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_ACL_POOLING_HPP
+#define CPU_AARCH64_ACL_POOLING_HPP
+
+#include "cpu/aarch64/acl_utils.hpp"
+#include "cpu/cpu_pooling_pd.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+struct acl_pooling_obj_t {
+    arm_compute::NEPoolingLayer pool;
+    arm_compute::Tensor src_tensor;
+    arm_compute::Tensor dst_tensor;
+};
+
+struct acl_pooling_conf_t {
+    bool is_max_pool;
+    arm_compute::PoolingLayerInfo pool_info;
+    arm_compute::TensorInfo src_info;
+    arm_compute::TensorInfo dst_info;
+};
+
+struct acl_pooling_resource_t : public resource_t {
+    acl_pooling_resource_t()
+        : acl_pooling_obj_(utils::make_unique<acl_pooling_obj_t>()) {}
+
+    status_t configure(const acl_pooling_conf_t &app) {
+        if (!acl_pooling_obj_) return status::out_of_memory;
+
+        // Init Compute Library tensors based on info from descriptor
+        acl_pooling_obj_->src_tensor.allocator()->init(app.src_info);
+        acl_pooling_obj_->dst_tensor.allocator()->init(app.dst_info);
+
+        acl_pooling_obj_->pool.configure(&acl_pooling_obj_->src_tensor,
+                &acl_pooling_obj_->dst_tensor, app.pool_info);
+
+        return status::success;
+    }
+
+    acl_pooling_obj_t &get_acl_obj() const { return *acl_pooling_obj_; }
+
+    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_pooling_resource_t);
+
+private:
+    std::unique_ptr<acl_pooling_obj_t> acl_pooling_obj_;
+}; // acl_pooling_resource_t
+
+struct acl_pooling_fwd_t : public primitive_t {
+    struct pd_t : public cpu_pooling_fwd_pd_t {
+        using cpu_pooling_fwd_pd_t::cpu_pooling_fwd_pd_t;
+        pd_t(const pooling_v2_desc_t *adesc, const primitive_attr_t *attr,
+                const pooling_fwd_pd_t *hint_fwd_pd)
+            : cpu_pooling_fwd_pd_t(adesc, attr, hint_fwd_pd), app() {}
+
+        DECLARE_COMMON_PD_T("acl", acl_pooling_fwd_t);
+
+        status_t init(engine_t *engine) {
+            bool ok = set_default_params() == status::success
+                    && is_fwd() // ACL supports forward propagation only
+                    && utils::everyone_is(data_type::f32, src_md()->data_type,
+                            dst_md()->data_type)
+                    && attr()->has_default_values()
+                    && attr_.set_default_formats(dst_md(0)) == status::success
+                    && !is_dilated() && !has_zero_dim_memory();
+            if (!ok) return status::unimplemented;
+
+            const pooling_v2_desc_t *pod = desc();
+
+            const bool is_training
```

Comment:
Is this a dead code? Workspace is not used at the execution (src/cpu/aarch64/acl_pooling.cpp)

Reply:
Removing the `init_default_ws()` call causes the tests `cpu-cnn-training-f32-cpp` and `cpu-cnn-training-f32-c` to fail. I'm not sure why yet, do you have any thoughts?

Reply:
The above tests fail fue to a problem where acl pooling is used for the forward training then ref is used for the backwards pass. during the backward pass there is a check on the workspace tensor (passed from the forward pass as a hint):
```
		init_default_ws();
                if (!compare_ws(hint_fwd_pd_)) return status::unimplemented;
```
Upon further investigation, we decided that all max forward training cases (when a workspace tensor would be needed) should be skipped, so this has been removed.